feather/server/src/systems/gamemode.rs

```diff
@@ -0,0 +1,200 @@
+                }
+                if **instabreak {
+                    instabreak_changes.push((entity, false));
+                    instabreak.0 = false;
+                }
+                if !**may_fly {
+                    may_fly_changes.push((entity, true));
+                    may_fly.0 = true;
+                }
+                if **may_build {
+                    build_changes.push((entity, false));
+                    may_build.0 = false;
+                }
+                if !**invulnerable {
+                    invulnerability_changes.push((entity, true));
+                    invulnerable.0 = true;
+                }
+            }
+            Gamemode::Survival => {
+                if **is_flying {
+                    fly_changes.push((entity, false));
+                    is_flying.0 = false;
+                }
+                if **instabreak {
+                    instabreak_changes.push((entity, false));
+                    instabreak.0 = false;
+                }
+                if **may_fly {
+                    may_fly_changes.push((entity, false));
+                    may_fly.0 = false;
+                }
+                if !**may_build {
+                    build_changes.push((entity, true));
+                    may_build.0 = true;
+                }
+                if **invulnerable {
+                    invulnerability_changes.push((entity, false));
+                    invulnerable.0 = false;
+                }
+            }
+            Gamemode::Adventure => {
+                if **is_flying {
+                    fly_changes.push((entity, false));
+                    is_flying.0 = false;
+                }
+                if **instabreak {
+                    instabreak_changes.push((entity, false));
+                    instabreak.0 = false;
+                }
+                if **may_fly {
+                    may_fly_changes.push((entity, false));
+                    may_fly.0 = false;
+                }
+                if **may_build {
+                    build_changes.push((entity, false));
+                    may_build.0 = false;
+                }
+                if **invulnerable {
+                    invulnerability_changes.push((entity, false));
+                    invulnerable.0 = false;
+                }
+            }
+        }
+        server
+            .clients
+            .get(client_id)
+            .unwrap()
+            .change_gamemode(**event);
+        server
+            .clients
+            .get(client_id)
+            .unwrap()
+            .send_abilities(&PlayerAbilities {
+                walk_speed,
+                fly_speed,
+                may_fly: *may_fly,
+                is_flying: *is_flying,
+                may_build: *may_build,
+                instabreak: *instabreak,
+                invulnerable: *invulnerable,
+            });
+    }
+    for (entity, flying) in fly_changes {
+        if flying {
```

Comment:
This if statement is redundant, just create `CreativeFlyingEvent` with the value of `flying`?